src/array_api_stubs/_draft/array_object.py

```diff
@@ -1031,9 +1073,10 @@ def __xor__(self: array, other: Union[int, bool, array], /) -> array:
         .. note::
            Element-wise results must equal the results returned by the equivalent element-wise function :func:`~array_api.bitwise_xor`.
         """
+        ...
 
     def to_device(
-        self: array, device: Device, /, *, stream: Optional[Union[int, Any]] = None
+        self: array, device: "Device", /, *, stream: int | Any | None = None  # type: ignore[type-var]
```

Comment:
There's been some minor discussion about this before (xref https://github.com/numpy/numpy/pull/19083#discussion_r638069860), but considering this class is now becoming a fully fledged protocol and not just a documentation device the `int` type in the `stream` annotation is starting to become a bit problematic as it makes it mandatory for any (concrete) subtype to support it (while this is, in fact, optional).

Reply:
A good point. I'm trying to translate this class, not change the spec, so I think this is best done in another PR.

Reply:
IMO this would not be a spec violation as it'd merely remove an inconsistency between the type signature as implied by the annotations `(..., stream: int | Any | None = ...)` versus the one described in docstring (`..., stream: None = ...`).

As a little bit more background on the nitty gritty part of the world of typing that's going on here: Contrary to the return type of callables (which is covariant and concrete protocol implementations can thus return any sub-type), argument types are contravariant, meaning that concrete implementations will either have to exactly match it _or_ return a supertype thereof (in the most extreme scenario that'd mean that even plain `object` is acceptable, though whether or not that's an actual good practice is another discussion entirely...).

What this means is that in order to proper subtype something like `def (stream: int | None = ...) -> object: ...` then concrete implementation _must_ support both `int` and `None`, something that contracts the actual docstring.

Reply:
Done!